gobblin-utility/src/main/java/gobblin/broker/iface/SharedResourceFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.broker.iface;
+
+/**
+ * A superclass of {@link ScopedSharedResourceFactory} and {@link UnscopedSharedResourceFactory}.
+ * {@link SharedResourceFactory}s create objects that a {@link GobblinBroker} can manage. The factory is required to be stateless.
+ * @param <T> The type of objects built by this factory.
+ * @param <K> The key used to create objects by this factory.
+ */
+public interface SharedResourceFactory<T, K extends SharedResourceKey> {
```

Comment:
This interface is strange. It does not define any "factory" methods and does not use any of the generic parameters.